nshlib/nsh_parse.c

```diff
@@ -2733,16 +2739,12 @@ static int nsh_parse_command(FAR struct nsh_vtbl_s *vtbl, FAR char *cmdline)
           if (param.fd_in != -1)
             {
               close(param.fd_in);
-              param.fd_in = -1;
               vtbl->np.np_redir_in = redirect_in_save;
             }
 
-          if (param.fd_out != -1)
```

Comment:
why skip the check

Reply:
> why skip the check

As line 2724/2730 shown, param.fd_out never be little than zero, because only reach here when `pipe2()`(line 2716/2722) returns success.